src/kpro_connection.erl

```diff
@@ -469,11 +496,29 @@ handle_msg({From, stop}, #state{mod = Mod, sock = Sock}, _Debug) ->
   Mod:close(Sock),
   maybe_reply(From, ok),
   ok;
+handle_msg(sasl_reauthenticate, State, Debug) ->
+  do_sasl_reauthenticate(State),
+  ?MODULE:loop(State, Debug);
 handle_msg(Msg, #state{} = State, Debug) ->
   error_logger:warning_msg("[~p] ~p got unrecognized message: ~p",
                           [?MODULE, self(), Msg]),
   ?MODULE:loop(State, Debug).
 
+do_sasl_reauthenticate(#state{client_id = ClientId, mod = Mod, sock = Sock, remote = {Host, _Port}, api_vsns = Versions, config = Config}) ->
+  %% Imitates logic in init -> connect, but using existing api_vsns and socket
+  Timeout = get_connect_timeout(Config),
+  Deadline = deadline(Timeout),
+  SaslOpts = get_sasl_opt(Config),
+  HandshakeVsn = case Versions of
+                   #{sasl_handshake := {_, V}} -> V;
+                   _ -> 0
+                 end,
+  ok = setopts(Sock, Mod, [{active, false}]), 
```

Comment:
before doing so. we should probably have to add a wait and drain all the inflight requests first ?
otherwise the kpro_sasl:auth may receive the response for previous requests and take it as an auth reply.

Reply:
That may indeed be the case. I will think about this a bit, and do some testing to verify that draining is necessary. I did not observe any draining in the reference Java client: https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/network/KafkaChannel.java#L602

Reply:
I did not study java code. but the problem for us is, the auth procedure enters a different request/response correlation (or rather the auth packets has no correlation at all).

in theory, since the auth backend is plugable, it might even enter a different framing mode
--- although currently it seems all auth mechanisms use packet-4 framing, which makes things easier.